packages/api/src/models/animalUseModel.js

```diff
@@ -0,0 +1,70 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import knex from '../util/knex.js';
+import baseModel from './baseModel.js';
+
+class AnimalUse extends baseModel {
+  static get tableName() {
+    return 'animal_use';
+  }
+
+  static get idColumn() {
+    return 'id';
+  }
+
+  // Optional JSON schema. This is not the database schema! Nothing is generated
+  // based on this. This is only used for validation. Whenever a model instance
+  // is created it is checked against this schema. http://json-schema.org/.
+  static get jsonSchema() {
+    return {
+      type: 'object',
+      required: ['key'],
+      properties: {
+        id: { type: 'integer' },
+        key: { type: 'string' },
+      },
+      additionalProperties: false,
+    };
+  }
+
+  static async getAnimalUsesForTypes() {
+    const typeUseRelationships = await knex('animal_type_use_relationship')
+      .select({
+        default_type_id: 'default_type_id',
+        useId: 'animal_use.id',
+        useKey: 'animal_use.key',
+      })
+      .join('animal_use', 'animal_type_use_relationship.animal_use_id', '=', 'animal_use.id');
+
+    const usesPerType = typeUseRelationships.reduce((map, { default_type_id, useId, useKey }) => {
+      map[default_type_id] = map[default_type_id] || [];
+      map[default_type_id].push({ id: useId, key: useKey });
+
+      return map;
+    }, {});
+
+    const response = Object.entries(usesPerType).map(([defaultTypeId, uses]) => {
```

Comment:
I was just curious about this transform. If we are going to be using it somewhat like a lookup table on the frontend, would it not be better to keep it as a true lookup table with keys as `default_type_id` (the null key could be `''`)?

Is it to be consistent in always returning arrays from our endpoints?

Reply:
Thank you for asking!!
I went back and forth to settle on the decision, but I wasn't even sure if it was a good idea to make the response like that...

I ended up going with an array because keys like "1", "2" or "" would make it difficult to understand what they represent at a glance. I researched which approach might be better, but unfortunately, I couldn't find any resources...😞 Returning the lookup object would definitely be more efficient, so if we are okay with the structure and doing something new, I would like to give it a try. It would be great if we could discuss this further on Monday!

Reply:
Hmmm! I like your readability argument so much I would very much like to keep it as is 🙏 The keys as `default_type_ids` would indeed become really cryptic as soon as we are a bit removed from this context.